de.fu_berlin.inf.dpp.whiteboard/src/de/fu_berlin/inf/dpp/whiteboard/ui/browser/BrowserSXEBridge.java

```diff
@@ -0,0 +1,96 @@
+package de.fu_berlin.inf.dpp.whiteboard.ui.browser;
+
+import org.apache.log4j.Logger;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+import de.fu_berlin.inf.dpp.whiteboard.net.WhiteboardManager;
+import de.fu_berlin.inf.dpp.whiteboard.sxe.ISXEMessageHandler.MessageListener;
+import de.fu_berlin.inf.dpp.whiteboard.sxe.SXEController;
+import de.fu_berlin.inf.dpp.whiteboard.sxe.net.SXEMessage;
+import de.fu_berlin.inf.dpp.whiteboard.sxe.records.ElementRecord;
+
+/**
+ * BrowserSXEBridge is the java interface between the SXEController which
+ * contains all model data and the browser which contains the whiteboard and
+ * manipulates the data.
+ */
+public class BrowserSXEBridge {
+
+    private static final Logger LOG = Logger.getLogger(BrowserSXEBridge.class);
+
+    private static final Gson gson = new Gson();
+
+    private static Gson gsonOnlyExposed = new GsonBuilder()
+        .excludeFieldsWithoutExposeAnnotation().create();
+
+    private static final String JAVA_FUNCTION_NAME = "__java_triggerAction";
+
+    private final SXEController sxe;
+
+    private final IWhiteboardBrowser browser;
+
+    private final BrowserActionExecuter executer;
+
+    /**
+     * Creates the action executor and initialises the message listener and the
+     * browser function
+     * 
+     * @param sxe
+     *            SXEController created in {@link WhiteboardManager}
+     * @param browser
+     *            the browser created in the view
+     */
+    public BrowserSXEBridge(SXEController sxe, IWhiteboardBrowser browser) {
+        this.sxe = sxe;
+        this.browser = browser;
+        executer = new BrowserActionExecuter(sxe);
+        sxe.addMessageListener(messageListener);
```

Comment:
I would add all code beginning from this line to a seperate init method instead. It is not always the best design to add listeners to objects in the CTOR.

Reply:
done